### ISSUE TITLE
Window 'resize' event fires twice

### ISSUE THREAD

**Issue by jportoles** [nwjs/nw.js#5678]
Version: 0.20.0-sdk
OS: Windows 10

I'm not sure if this is a bug or just undocumented behavior, but reporting just in case. When resizing a window via resizeTo/resizeBy methods, the 'resize' event emitted by the NW.js Window object (as documented [here](http://docs.nwjs.io/en/latest/References/Window/#event-resize)) fires twice, once for changed width, and another for changed height. Usually one would expect the resize event to fire only once both dimensions are set, but if this intended it should at least be mentioned in the documentation.

**Comment by Christywl** [comment]
Reproduced on Linux and Windows with nwjs-sdk-0.21.0, but there is a little difference on Linux and Windows.
1) Windows: the 'resize' event fired twice, once for width change, and another for height change
![2](https://cloud.githubusercontent.com/assets/20039821/23601236/6fe65002-0285-11e7-80be-f1b9ebda2847.png)

2) Linux: the 'resize' event fired fourth times, the first/third times for width change, and the second/fourth times for height change
![3](https://cloud.githubusercontent.com/assets/20039821/23601261/8afa2e04-0285-11e7-9634-c506a1fa9e00.png)


My code:
```
<!DOCTYPE html>
<html>
<body>
<button id="resize">resize by (100, 100)</button>
<div id="log"></div>
<script type="text/javascript">
  var win = nw.Window.get();
  document.getElementById('resize').addEventListener('click', () => {
    win.resizeBy(100, 100);
    win.on('resize', (width, height) => {
      document.getElementById('log').innerHTML += `size from resize event: width is ${width}, height is ${height} <br>`; 
	});
   setTimeout(() => {	win.removeAllListeners('resize');},1000);
});

</script>

</body>
</html>
```

**Comment by xzhan96** [comment]
@rogerwang I'd like to investigate this issue, can I change the owner to me?

**Comment by xzhan96** [comment]
@jportoles 
I have investigated this issue, this is not bug, please see the same question [here](http://stackoverflow.com/questions/15812618/window-onresize-fires-twice).

@Christywl for the issue that 'resize' event fired fourth times, it will also be fixed by #5748.

For the above sample, you can change the code below
```
win.on('resize', (width, height) => {
      document.getElementById('log').innerHTML += `size from resize event: width is ${width}, height is  ${height} <br>`; 
});
```
to
```
win.on('resize', (width, height) => {
  if (res){clearTimeout(res)};
  res = setTimeout(function(){
  document.getElementById('log').innerHTML += `size from resize event: width is ${width}, height is ${height} <br>`;},100);
});
```
**And if with the patch #5748, everything works well:** 

![screenshot from 2017-03-08 14 29 14](https://cloud.githubusercontent.com/assets/13726257/23692642/e66d0a3c-040a-11e7-80b8-241011cdcac7.png)

**Comment by xzhan96** [comment]
The issue that 'resize' event fired fourth times will also be fixed by #5748.
